src/Logic.hs

```diff
@@ -436,26 +436,29 @@ handleCommentAdded triggerConfig projectConfig mergeWindowExemption prId author
       day <- dayOfWeek . utctDay <$> getDateTime
       if isApproved
```

Comment:
`isApproved` will only be `True` if `approvalType` is a `Just`, so these `Nothing` cases can never occur.

The place we interpret the command is in https://github.com/channable/hoff/blob/master/src/Logic.hs#L376-L398 We return a `Nothing` over there, which causes the `approvalType` to be `Nothing`, as you correctly interpreted. You probably can revert the code and leave  a comment on the `else` branch.

Reply:
Oh, right. Thanks! I'll go fix it :sweat_smile:

Reply:
Simply putting it in the else branch made a test fail because it was expecting no reply to a message that did not mention the bot, which seems like intended behavior to me.

I ended up creating a result type for the parsing function that specifies why it failed to parse (invalid prefix or invalid command) and pattern match on that result in order to avoid this issue. It does not reply when the author is not a reviewer, because that might cause abuse from unauthorized users.